README.md

```diff
@@ -1,24 +1,107 @@
 
 Inside `my-plugin` you can find the files that defines the resources to be exposed by the plugin to be used by the burner wallet in order to interact with the ERC677 to ERC677 bridge extension:
-- `Stake` - extends from `ERC677Asset` defined in `tokenbridge-plugin`
-- `xStake` - extends from `ERC677Asset` defined in `tokenbridge-plugin`
-- `StakeBridge` - extends from `Mediator` defined in `tokenbridge-plugin`
+- `Stake` - extends from `ERC677Asset` defined in `@poanet/tokenbridge-bw-exchange`
+- `xStake` - extends from `ERC677Asset` defined in `@poanet/tokenbridge-bw-exchange`
+- `StakeBridge` - extends from `Mediator` defined in `@poanet/tokenbridge-bw-exchange`
+
+The class `ERC677Asset` extends from `ERC20Asset` and overrides some methods:
+- `_send`: In this case the method `transferAndCall` is used instead of the `transfer` method.
+- `startWatchingAddress`: Add logic to correctly track the transfer events related to the wallet account.
+- `getTx`: Overrides the logic to return the stored information about a transfer transaction to avoid errors on the information displayed.
+
+The class `Mediator` extend from some other classes that has `Pair` as base. It implements the previous mentioned methods to estimate and exchange the assets. It also implements some methods to detect the exchange finalization.
+Here are some details of the implemented methods:
+- `estimateAtoB`: Get the fee and calculates the amount of B Asset the user will receive.
+- `estimateBtoA`: Get the fee and calculates the amount of A Asset the user will receive.
+- `exchangeAtoB`: Sends A Asset to the mediator contract. Then it calls `detectExchangeAToBFinished`.
+- `exchangeBtoA`: Sends B Asset to the mediator contract. Then it calls `detectExchangeBToAFinished`.
+- `detectExchangeAToBFinished`: Wait for events emitted by the mediator contract to detect the exchange finalization.
+- `detectExchangeBToAFinished`: Wait for events emitted by the mediator contract to detect the exchange finalization.
+
+In this repo example case, the class `StakeBridge` extends the `Mediator` class but has some differences on how to estimate the values and calculate the transfers fees.
+That's why it overrides the methods:
+- `estimateAtoB`: To add specific logic to get the fee from the A to B exchange and estimate the value that will be received.
+- `estimateBtoA`: To avoid fee calculations since there are no fees for B to A exchange.
 
 You can extend or replace these resources based on your use case.
 
+Inside the wallet folder, the file `wallet/src/index.tsx` integrates the plugin into the wallet by importing the resources in the following line:
+
+```
+import { Stake, xStake, StakeBridge } from 'my-plugin'
+```
+
+To be able to use and to display the balances for Stake and xStake it is necessary to list them in the list of assets as follow:
+```
+const core = new BurnerCore({
+    ...
+  assets: [Stake, xStake]
```

Comment:
Please also specify some where about gateways that are used to automatically choose a Web3 provider. Describe shortly a way how the provider is being chosen.